lib/langserver.js

```diff
@@ -0,0 +1,58 @@
+const { URI } = require('vscode-uri');
+const { promises: fs } = require('fs');
+const { hasOctaneImports, transformOctaneImports } = require('./octane-utils');
+const { transformImports } = require('./utils');
+
+async function onComplete(projectRoot, opts) {
+  function log() {
+    return opts.server.connection.console.log(...arguments);
+  }
+
+  if (opts.focusPath.node.type === 'ElementNode') {
+    let filePath = uriToPath(opts.textDocument.uri);
+
+    if (filePath) {
+      let content = await fs.readFile(filePath, 'utf-8');
+      let mightHaveImports = content.includes('import');
+      if (!mightHaveImports) {
+        return [];
+      }
+      let isOctane = hasOctaneImports(content);
+      let componentImports;
+
+      if (isOctane) {
+        let legacyContent = transformOctaneImports(content, filePath);
```

Comment:
Any way to get the file path without having to keep `uriToPath`? Looked at the source but didn't see anything like that

Reply:
@lifeart

Reply:
yeah, you have to use it, I'm not confident about including it to public API, but it discussable